packages/dart_firebase_admin/lib/src/app/credential.dart

```diff
@@ -41,11 +41,20 @@ class Credential {
   }
 
   /// Log in to firebase using the environment variable.
-  Credential.fromApplicationDefaultCredentials({String? serviceAccountId})
-      : this._(
-          null,
-          serviceAccountId: serviceAccountId,
-        );
+  factory Credential.fromApplicationDefaultCredentials({
+    String? serviceAccountId,
+  }) {
+    ServiceAccountCredentials? creds;
+    final maybeConfig = Platform.environment['GOOGLE_APPLICATION_CREDENTIALS'];
+    if (maybeConfig != null && maybeConfig.startsWith('{')) {
```

Comment:
I don't like this random `startWith('{`

I'd rather use `try { jsonDecode } on FormatException {...}`

Reply:
Alright, good suggestion.

Reply:
Agree, although that's the implementation in node sdk so probably taken from there